src/maintainer.rs

```diff
@@ -107,16 +107,30 @@ impl TableMaintainer {
             tokio::time::sleep(idle_time).await;
 
             info!("TableMaintainer::monitor_buckets woke up");
-            self.ping_idle_buckets().await;
+            self.find_new_nodes().await;
 
             info!("TableMaintainer::monitor_buckets removing idle nodes");
-            self.ktable.write().await.remove_idle_nodes();
+            self.ping_and_remove_idles().await;
         }
     }
 
     /// Search for idle buckets (no message received) and try to contact some of
     /// the belonging nodes
-    async fn ping_idle_buckets(&self) {
+    async fn ping_and_remove_idles(&self) {
+        let idles = self
+            .ktable
+            .read()
+            .await
+            .idle_nodes()
+            .map(|n| *n.value().address())
+            .collect();
+        self.send((Message::Ping(self.header), idles)).await;
```

Comment:
Perhaps I am mistaken, but I find it a bit confusing that the Maintainer sends a message to multiple idle nodes with a `send` method. A better name would be `broadcast`.

Reply:
`broadcast` send a message to the whole network
`send` only to the selected addresses